### PR TITLE
Added support for overlay rotations in Google Maps API

### PR DESCRIPTION
**Pull request by ArnoldoJr**
### Does any other open PR do the same thing?
No

### What issue is this PR fixing?

[2892](https://github.com/react-native-community/react-native-maps/issues/2892)
[2850](https://github.com/react-native-community/react-native-maps/issues/2850)

### How did you test this PR?

- Created plain `MapView`
- Added an `Overlay` child view to it:
- `Android:` Verified that both methods (bounds vs location + bearing + anchor) work
- `iOS:` Verified that setting `bearing` to something different from zero, rotates the overlay.

### CODE REVIEW

File: docs/overlay.md [hunk 1]
```diff
@@ -6,7 +6,12 @@
 |---|---|---|---|
 | `image` | `ImageSource` | A custom image to be used as the overlay. Only required local image resources and uri (as for images located in the net) are allowed to be used.
 | `bounds` | `Array<LatLng>` |  | The coordinates for the image (left-top corner, right-bottom corner). ie.```[[lat, long], [lat, long]]```
+| `bearing` | `Float` | `0` | `Google Maps API only` The bearing in degrees clockwise from north. Values outside the range [0, 360) will be normalized.
 | `tappable` | `Bool` | `false` | `Android only` Boolean to allow an overlay to be tappable and use the onPress function.
+| `location` | `[Float, Float]` |  | `Android only` Location of the image overlay.
```
Comment (Trias) [inline docs/overlay.md]: This is a LatLng according to the code

Comment (ArnoldoJr) [reply docs/overlay.md]: [LatLng](https://github.com/react-native-community/react-native-maps/blob/bac105efbeee43c335ae81082b054ca43447fbea/index.d.ts#L18) is is an object with two fields :
```
export interface LatLng {
   latitude: number;
   longitude: number;
}
```
 
The `location` prop that we are passing in javascript requires a [Coordinate](https://github.com/react-native-community/react-native-maps/blob/bac105efbeee43c335ae81082b054ca43447fbea/index.d.ts#L478) object:

`type Coordinate = [number, number];`

which is pretty much an array of two numbers.

So we are actually passing a `Coordinate`, not a `LatLng`.

**Comment by geekazoid-at** [comment]
Any chance we could get the conflicts fixed and this PR merged? I'd really want this feature too.

**Comment by SaeedZhiany** [comment]
a

**Comment by the-habu** [comment]
Can anyone help me? I'd like to checkout the branch to contribute/rebase/fix-the-conflicts/etc but I can't seem to find where it is hosted.

Is it long gone and @ArnoldoJr removed it? Thank you in advance!

**Comment by christopherdro** [comment]
Superseded by #3737

**Pull request closed without merging**